src/server/server.ts

```diff
@@ -242,6 +287,183 @@ export class CTProtoServer<AuthRequestPayload, AuthData, ApiRequest extends NewM
     }
   }
 
+  /**
+   * Process buffer message.
+   *
+   * @param client - connected client
+   * @param message - accepted message
+   */
+  private async handleBufferMessage(client: Client<AuthData, ApiResponse, ApiUpdate>, message: Buffer): Promise<void> {
+    /**
+     * Parsing meta data from buffer message
+     */
+    const fileIdLength = 10;
+    const chunkNumberOffset = 10;
+    const sizeOffset = 14;
+    const sizeDataLength = 4;
+
+    const fileId = message.slice(0, fileIdLength).toString();
+    const chunkNumber = message.readInt32LE(chunkNumberOffset);
+    const size = message.readInt32LE(sizeOffset);
+
+    /**
+     * Getting file data
+     */
+    const dataOffset = sizeOffset + sizeDataLength;
+    const fileChunk = message.slice(dataOffset, dataOffset + size);
+
+    /**
+     * Parsing payload message in buffer message
+     */
+    const strPayload = message.slice(dataOffset + size).toString();
+
+    const payload = JSON.parse(strPayload);
+
+    const chunkOffset = size * chunkNumber;
+
+    let file = this.uploadingFiles.find((req) => req.id === fileId);
+
+    /**
+     * Metadata of the first chunk includes additional information
+     */
+    if ( !file ) {
+      /**
+       * Create Buffer for file data
+       */
+      const fileData = Buffer.alloc(fileChunk.length + chunkOffset);
+
+      fileChunk.copy(fileData, chunkOffset);
+
+      /**
+       * Create new file object, the first chunk has more info about file ( chunks, payload, type of request )
+       */
+      if (chunkNumber === 0) {
+        this.uploadingFiles.push( { id: fileId,
+          uploadedChunks: [],
+          file: fileData,
+          chunks: payload.chunks,
+          payload: payload.payload,
+          type: payload.type,
+        } );
+      } else {
+        this.uploadingFiles.push( { id: fileId,
+          uploadedChunks: [],
+          file: fileData,
+        } );
+      }
+
+      file = this.uploadingFiles.find((req) => req.id === fileId);
```

Comment:
do we need to `push` and then `find`? Why not to store object at the variable first, then push.